src/main/java/in/projecteka/consentmanager/clients/IdentityServiceClient.java

```diff
@@ -0,0 +1,52 @@
+package in.projecteka.consentmanager.clients;
+
+import in.projecteka.consentmanager.clients.properties.IdentityServiceProperties;
+import in.projecteka.consentmanager.clients.model.KeycloakCreateUserRequest;
+import in.projecteka.consentmanager.clients.model.KeycloakToken;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+public class IdentityServiceClient {
+
+    private final WebClient.Builder webClientBuilder;
+
+    public IdentityServiceClient(WebClient.Builder webClientBuilder,
+                                 IdentityServiceProperties identityServiceProperties) {
+        this.webClientBuilder = webClientBuilder;
+        this.webClientBuilder.baseUrl(identityServiceProperties.getBaseUrl());
+    }
+
+    public Mono<?> createUser(KeycloakToken keycloakToken, KeycloakCreateUserRequest request) {
+        String accessToken = String.format("Bearer %s", keycloakToken.getAccessToken());
+        return webClientBuilder.build()
+                .post()
+                .uri(uriBuilder ->
+                        uriBuilder.path("/admin/realms/consent-manager/users").build())
```

Comment:
can we take this out to application property?

Reply:
Technically yes, but I would prefer not to do. Because if the URL changes, most likely we have to  change the body/response of the request too, and this client implementation is specific to Keycloak.